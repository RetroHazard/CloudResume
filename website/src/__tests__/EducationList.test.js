import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import EducationList from '../components/education_list';

// Sample JSON data
const sampleData = {
    Education: [
        {
            school: "Sample University",
            degree: "BSc. Sample Degree",
            location: "Sample City, Country",
            category: "Sample Category",
            start: "2000",
            end: "2004",
            logo: "./images/placeholder.png",
            details: [
                "Sample detail 1",
                "Sample detail 2",
                "Sample detail 3",
                "Sample detail 4"
            ],
            links: [
                {
                    program: [
                        {
                            website: "https://sampleprogram.com",
                            icon: "fa6-solid:circle-info"
                        }
                    ],
                    school: [
                        {
                            website: "https://sampleuniversity.com",
                            icon: "fa6-solid:globe"
                        }
                    ]
                }
            ]
        }
    ]
};

// Mock the DataLoader component
jest.mock('../utils/dataLoader', () => ({
    __esModule: true,
    default: ({ children }) => children(sampleData),
}));

describe('EducationList Component', () => {
    it('renders correctly', () => {
        const { getByText, getByAltText, getAllByRole } = render(<EducationList />);

        // Check for education details
        expect(getByText('Sample University')).toBeInTheDocument();
        expect(getByText('BSc. Sample Degree')).toBeInTheDocument();
        expect(getByText('Sample Category')).toBeInTheDocument();
        expect(getByText('2000 - 2004')).toBeInTheDocument();
        expect(getByText('Sample City, Country')).toBeInTheDocument();
        expect(getByAltText('Sample University Logo')).toBeInTheDocument();

        // Check for details
        sampleData.Education[0].details.forEach(detail => {
            expect(getByText(detail)).toBeInTheDocument();
        });

        // Check for links
        const links = getAllByRole('link');
        expect(links).toHaveLength(2); // Expecting 2 links

        expect(links[0]).toHaveAttribute('href', 'https://sampleprogram.com');
        expect(links[1]).toHaveAttribute('href', 'https://sampleuniversity.com');
    });

    it('matches snapshot', () => {
        const { container } = render(<EducationList />);
        expect(container).toMatchSnapshot();
    });
});
