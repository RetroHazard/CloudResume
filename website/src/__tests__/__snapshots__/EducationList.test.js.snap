// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducationList Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="flex flex-col gap-3"
    >
      <div
        class="flex w-full justify-between gap-2"
      >
        <div
          class="flex gap-4"
        >
          <img
            alt="Sample University Logo"
            class="hidden h-[7rem] w-[7rem] rounded-xl sm:block"
            src="./images/placeholder.png"
          />
          <div
            class="flex flex-col"
          >
            <h3
              class="h3 mb-0 font-extrabold text-content-title max-sm:text-sm sm:text-xs md:text-base"
            >
              Sample University
            </h3>
            <p
              class="mb-0.5 font-semibold leading-snug text-content-subtitle max-sm:text-sm sm:text-xs md:text-base"
            >
              BSc. Sample Degree
            </p>
            <p
              class="mb-0 font-medium leading-snug text-content-accent max-sm:text-xs sm:text-xs md:text-sm"
            >
              Sample Category
            </p>
            <p
              class="mb-0 font-medium leading-snug text-content-date max-sm:text-xs sm:text-xs md:text-sm"
            >
              2000
               - 
              2004
            </p>
            <p
              class="mb-0 font-light leading-snug text-content-accent max-sm:text-xs sm:text-xs md:text-xs"
            >
              Sample City, Country
            </p>
          </div>
        </div>
        <div
          class="flex flex-wrap gap-3 max-sm:flex-col sm:flex-col"
          id="Sample University Links"
        >
          <a
            aria-label="program"
            class="social-link"
            href="https://sampleprogram.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              height="1.25em"
              icon="fa6-solid:circle-info"
              width="1.25em"
            />
          </a>
          <a
            aria-label="school"
            class="social-link"
            href="https://sampleuniversity.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              height="1.25em"
              icon="fa6-solid:globe"
              width="1.25em"
            />
          </a>
        </div>
      </div>
      <div
        class="mb-3 font-normal leading-relaxed max-sm:text-xs sm:text-xs sm:leading-relaxed md:text-sm"
      >
        <ul
          class="list-disc pl-10 pr-5"
        >
          <li>
            Sample detail 1
          </li>
          <li>
            Sample detail 2
          </li>
          <li>
            Sample detail 3
          </li>
          <li>
            Sample detail 4
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
